packages/dev/core/src/Misc/tools.ts

```diff
@@ -45,6 +45,39 @@ export class Tools {
         FileToolsOptions.BaseUrl = value;
     }
 
+    /**
+     * This function checks whether a URL is absolute or not.
+     * It will also detect data and blob URLs
+     * @param url the url to check
+     * @returns is the url absolute or relative
+     */
+    public static IsAbsoluteUrl(url: string): boolean {
+        return url.indexOf("://") !== -1 || url.indexOf("//") === 0 || url.indexOf("data:") === 0 || url.indexOf("blob:") === 0;
+    }

```

Comment:
This is what I did locally, not sure if it's better or not:
```js
/**
 * Returns whether a URL is absolute or not.
 * @param url The URL to check
 * @returns True if the URL is absolute or false otherwise
 */
export function isAbsoluteUrl(url: string): boolean {
    // See https://stackoverflow.com/a/38979205.

    // URL is protocol-relative (= absolute)
    if (url.indexOf("//") === 0) {
        return true;
    }

    // URL has no protocol (= relative)
    if (url.indexOf("://") === -1) {
        return false;
    }

    // URL does not contain a dot, i.e. no TLD (= relative, possibly REST)
    if (url.indexOf(".") === -1) {
        return false;
    }

    // URL does not contain a single slash (= relative)
    if (url.indexOf("/") === -1) {
        return false;
    }

    // The first colon comes after the first slash (= relative)
    if (url.indexOf(":") > url.indexOf("/")) {
        return false;
    }

    // Protocol is defined before first dot (= absolute)
    if (url.indexOf("://") < url.indexOf(".")) {
        return true;
    }

    // Anything else must be relative
    return false;
}
```

Reply:
I can use that as well. it's missing data and blob, but i can add them.

Reply:
Done